models/DataObject/ClassDefinition/Data/Fieldcollections.php

```diff
@@ -448,13 +448,48 @@ public function preSetData(mixed $container, mixed $data, array $params = []): m
         return $data;
     }
 
-    /**
-     * @param DataObject\Concrete|null $object
-     *
-     */
-    public function getDataForGrid(?DataObject\Fieldcollection $data, Concrete $object = null, array $params = []): string
+    public function getDataForGrid(?DataObject\Fieldcollection $data, DataObject\Concrete $object = null, array $params = []): ?array
```

Comment:
not a big deal maybe but to avoid the return type covariance break

```suggestion
    public function getDataForGrid(?DataObject\Fieldcollection $data, DataObject\Concrete $object = null, array $params = []): string|array
    {
        if (null === $data) {
            return 'NO DATA';
        }
```